### DISCUSSION TITLE
Neorg in LazyVim

### DISCUSSION THREAD

**Discussion by PatzerZach** [nvim-neorg/neorg#1367] [Q&A]
Hello!

So I am very new to this whole, doing everything from the terminal, but so far I love it. I do have a question about installing this. I am running wsl on my windows machine and I am using Lazy Vim as my current NeoVim. I really want to be able to use Neorg to take notes for my college classes but I am having a hard time installing. To be honest, I am just confused, I really have no clue on how to install neorg and that have it work with my Lazy Vim. I want to be able to open Neorg in Lazy Vim so I can incorporate it in my work flow. Could I please get some help on how to fully install and implement Neorg in Lazy Vim. If someone could help me that would be awesome!!! Thank you so much!!!

**Comment by pysan3** [comment]
What did you try? What do you not understand?

Did you read https://github.com/pysan3/Norg-Tutorial?

**Reply by PatzerZach** [reply]
All ive tried to do was install it, but I can’t figure out what files I am supposed to put the installation code and plugin code into. Do I have to install a seperate package manager or something like lazy.nvim? I just don’t know how to install it because the instructions don’t have specifics of where to put it. I especially don’t know because I’m new to all of this. The YouTube videos I have found, including the one on Neorg’s GitHub page, don’t show you how to install it. They just say, “Once you’ve installed it”.

**Reply by pysan3** [reply]
Welcome aboard. 

I suppose you are using LazyVim so they have an awesome documentation that helps you install plugins. 

http://www.lazyvim.org/configuration/plugins

If you are a beginner and you are willing to learn neovim, ngl all pages there will be an amazing starting point. Read through all pages in your free time. 

http://www.lazyvim.org/

After reading the doc, I think when you open LazyVim, you are greeted with a welcome page. Press `c` to search thru the config files. Start typing _example_ and I think you can find `examples.lua`. 

You'll see real code examples of how to install plugins with lazy. Good luck.

**Comment by weberam2** [comment]
For me, I first needed to install lua5.1 and liblua5.1-0-dev on my system (outside neovim)

```
sudo apt install liblua5.1-0-dev
```

Then in LazyVim, uninstall both neorg and luarocks (press space+l to get to the lazyvim menu, find neorg and press 'x' and the same for luarocks

Now exit and go back in to LazyVim. Hopefully it'll install properly now.

Here is my neorg.lua file:

```
return {
  {
    "vhyrro/luarocks.nvim",
    priority = 1000,
    config = true,
  },
  {
    "nvim-neorg/neorg",
    dependencies = { "luarocks.nvim" },
    version = "*",
    -- config = true,
    config = function()
      require("neorg").setup({
        load = {
          ["core.defaults"] = {}, -- Loads default behaviour
          ["core.concealer"] = {}, -- Adds pretty icons to your documents
          ["core.ui.calendar"] = {},
          ["core.completion"] = { config = { engine = "nvim-cmp", name = "[Norg]" } },
          ["core.integrations.nvim-cmp"] = {},
          -- ["core.concealer"] = { config = { icon_preset = "diamond" } },
          ["core.esupports.metagen"] = { config = { type = "auto", update_date = true } },
          ["core.qol.toc"] = {},
          ["core.qol.todo_items"] = {},
          ["core.looking-glass"] = {},
          ["core.presenter"] = { config = { zen_mode = "zen-mode" } },
          ["core.export"] = {},
          ["core.export.markdown"] = { config = { extensions = "all" } },
          ["core.summary"] = {},
          ["core.tangle"] = { config = { report_on_empty = false } },
          ["core.dirman"] = { -- Manages Neorg workspaces
            config = {
              workspaces = {
                notes = "~/notes/notes",
                work = "~/notes/work",
              },
              default_workspace = "work",
            },
          },
        },
      })
    end,
  },
}

```

**Reply by PatzerZach** [reply]
Thank you! So I ran that first command that you gave me, but then when I went into lazy vim to uninstall both of those things, they didn't even show up. I don't think I have downloaded them yet. That is my main issue, I cant figure out how to install luarocks and neorg. What files am I supposed to be putting the installation code into for both luarocks and neorg? Or do I install them by using commands in the terminal? Do I have to install them in the init.lua file? If I could get some help on how to install both of these tools that would be awesome! I dont know if this changes anything, but this is all running in window's WSL. Also what file do I put plugins for lazy vim / neovim?

**Reply by evpeople** [reply]
Thank you so much!!!

**Comment by weberam2** [comment]
Happy to help

You should have a folder like:
~/.config/nvim/lua/plugins

Inside that folder, you should make a file called `neorg.lua` and populate it with something like:
```
return {
  {
    "vhyrro/luarocks.nvim",
    priority = 1000,
    config = true,
  },
  {
    "nvim-neorg/neorg",
    dependencies = { "luarocks.nvim" },
    version = "*",
    -- config = true,
    config = function()
      require("neorg").setup({
        load = {
          ["core.defaults"] = {}, -- Loads default behaviour
          ["core.concealer"] = {}, -- Adds pretty icons to your documents
          ["core.ui.calendar"] = {},
          ["core.completion"] = { config = { engine = "nvim-cmp", name = "[Norg]" } },
          ["core.integrations.nvim-cmp"] = {},
          -- ["core.concealer"] = { config = { icon_preset = "diamond" } },
          ["core.esupports.metagen"] = { config = { type = "auto", update_date = true } },
          ["core.qol.toc"] = {},
          ["core.qol.todo_items"] = {},
          ["core.looking-glass"] = {},
          ["core.presenter"] = { config = { zen_mode = "zen-mode" } },
          ["core.export"] = {},
          ["core.export.markdown"] = { config = { extensions = "all" } },
          ["core.summary"] = {},
          ["core.tangle"] = { config = { report_on_empty = false } },
          ["core.dirman"] = { -- Manages Neorg workspaces
            config = {
              workspaces = {
                notes = "~/notes/notes",
                work = "~/notes/work",
              },
              default_workspace = "work",
            },
          },
        },
      })
    end,
  },
}
```

I say something like, because maybe you don't want all the same things I load in there (["core.export.markdown"]) for example

Then load up neovim and you should see lazy install everything

Btw, the LazyVim.org website is really useful reading it through start to finish(ish)

Goodluck!

**Reply by PatzerZach** [reply]
Thank you!!! So I did everything you said and it is now installed!!! But now I cant figure out how to use it. How do I launch it? I have tried typing Neorg into the command line in lazy vim but it says it isnt a command. On Neorg's github page about installing with lazy they have two code snippets. Do I need to put this in two different files? Here is what the Neorg page says:

Ensure you have vhyrro/luarocks.nvim installed:

{
    "vhyrro/luarocks.nvim",
    priority = 1000,
    config = true,
}

Add the following to your plugin list:

{
    "nvim-neorg/neorg",
    dependencies = { "luarocks.nvim" },
    lazy = false, -- Disable lazy loading as some `lazy.nvim` distributions set `lazy = true` by default
    version = "*", -- Pin Neorg to the latest stable release
    config = true,
}

So do I need two different files? The neorg.lua file with that long snip you gave me , and then an installation file? Thank you again so much for your help!!! Here is what I am seeing right now on lazy:

![LAZY VIM POST](https://github.com/nvim-neorg/neorg/assets/92264586/63ae7a56-808f-45c4-8cb1-75abd81e08ce)

**Reply by weberam2** [reply]
I just have the one file that you see above.
